frontend/src/Routes/DynamicRoute/DynamicRoute.tsx

```diff
@@ -16,62 +16,51 @@ type DynamicRouteProps = {
-  const [, section] = pathname.split('/');
-  return matchPath(pathname, { path, exact }) || matchPath(pathname, { path: `/${section}${path}`, exact });
-};
-const DynamicRoute: React.FC<DynamicRouteProps> = ({ location }) => {
-  const [Component, setComponent] = React.useState<React.ExoticComponent<any>>(React.Fragment);
-  const [currClassName, setCurrClassName] = React.useState<string>(null);
-  // TODO: This seems to get executed when the tree fails to have PluginStoreProvider and then explodes
+type StateRoute = {
+  className: string;
+  Component: React.ExoticComponent<any>;
+}
+const DynamicRoute: React.FC<DynamicRouteProps> = () => {
+  const [routes, setRoutes] = React.useState<StateRoute[]>([]);
   const dynamicRoutePages = useExtensions<DynamicRoutePage>(isDynamicRoutePage);
+
   React.useEffect(() => {
-    if (location) {
-      const { properties: currRoute, pluginName } =
-        dynamicRoutePages.find(({ properties }) => {
-          if (Array.isArray(properties.path)) {
-            return properties.path.some((path) => checkPath(location.pathname, { ...properties, path }));
+    setRoutes(dynamicRoutePages.map(({ properties: { component, ...currRoute }, pluginName }) => {
+      return {
+        ...currRoute,
+        className: camelCase(pluginName),
+        Component: React.lazy(async () => {
+          try {
+            return {
+              default: (await component()) || Loader,
+            };
+          } catch (e) {
+            console.error(e);
+            return {
+              default: () => (
+                <Bullseye>
+                  <ErrorState errorTitle={`There was an error while loading ${pluginName} plugin.`} />
+                </Bullseye>
+              ),
+            };
           }
-          return checkPath(location.pathname, properties as RoutePage);
-        }) || {};
-      if (currRoute) {
-        setCurrClassName(camelCase(pluginName));
-        setComponent(() =>
-          React.lazy(async () => {
-            try {
-              return {
-                default: (await currRoute.component()) || Loader,
-              };
-            } catch (e) {
-              return {
-                default: () => (
-                  <Bullseye>
-                    <ErrorState errorTitle={`There was an error while loading ${pluginName} plugin.`} />
-                  </Bullseye>
-                ),
-              };
-            }
-          }),
-        );
+        }),
       }
-    }
-  }, [location, dynamicRoutePages]);
+    }))
+  }, [dynamicRoutePages]);
+
+  console.log(routes, 'this is routes');
 
   return (
-    <section className={currClassName || ''}>
-      {Component ? (
-        <React.Suspense fallback={null}>
+    <React.Suspense fallback={null}>
+      {routes.map(({ className, Component, ...currCoute }) => <React.Fragment>
+        <Route
+          {...currCoute}
+          render={() => <section className={className || ''}>
```

Comment:
```suggestion
          render={() => <section className={className}>
```

Reply:
This could lead to `undefined` className.

Reply:
React will just ignore undefined values set on a prop though so it should be fine

Reply:
Mhh I could swear I've seen `undefined` classes in the past. But can't reproduce it on this. Will apply the change.